cpg_utils/config.py

```diff
@@ -32,7 +32,38 @@ def set_config_path(config_path: str) -> None:
 def get_config() -> dict:
     """Returns the configuration dictionary.
 
-    Call set_config_path beforehand to override the default path.
+    Call `set_config_path` beforehand to override the default path.
+    See `read_config` for the path value semantics.
+
+    Notes
+    -----
+    Caches the result based on the config path alone.
+
+    Returns
+    -------
+    dict
+    """
+
+    global _config
+    if _config is None:  # Lazily initialize the config.
+        assert (
+            _config_path
+        ), 'Either set the CPG_CONFIG_PATH environment variable or call set_config_path'
+
+        _config = read_config(_config_path)
+
+        # Print the config content, which is helpful for debugging.
+        print(f'Configuration at {_config_path}:\n{toml.dumps(_config)}')
+
+    return _config
+
+
+def read_config(config_path: str) -> dict:
```

Comment:
Should it take a list, and the usage would be `read_config(_config_path.split(','))`?

Reply:
Yes, thanks, that's much better!